examples/luna/ntsb_loader.py

```diff
@@ -0,0 +1,148 @@
+            "aircraft": {"type": "string"},
+            "flightConductedUnder": {"type": "string"},
+            "accidentNumber": {"type": "string"},
+            "registration": {"type": "string"},
+            "injuries": {"type": "string"},
+            "aircraftDamage": {"type": "string"},
+        },
+    }
+    doc.properties.update({"_schema": schema_json, "_schema_class": "Flight Accident Report"})
+    return doc
+
+
+def convert_timestamp(doc: Document) -> Document:
+    if "dateAndTime" not in doc.properties["entity"] and "dateTime" not in doc.properties["entity"]:
+        return doc
+    raw_date: str = doc.properties["entity"].get("dateAndTime") or doc.properties["entity"].get("dateTime")
+    raw_date = raw_date.replace("Local", "")
+    parsed_date = parser.parse(raw_date, fuzzy=True)
+    extracted_date = parsed_date.date()
+    doc.properties["entity"]["day"] = extracted_date.isoformat()
+    if parsed_date.utcoffset():
+        doc.properties["entity"]["isoDateTime"] = parsed_date.isoformat()
+    else:
+        doc.properties["entity"]["isoDateTime"] = parsed_date.isoformat() + "Z"
+
+    return doc
+
+
+if os.path.exists("/.dockerenv"):
+    opensearch_host = "opensearch"
+    print("Assuming we are in a Sycamore Jupyter container, using opensearch for OpenSearch host")
+else:
+    opensearch_host = "localhost"
+    print("Assuming we are running outside of a container, using localhost for OpenSearch host")
+
+os_client_args = {
+    "hosts": [{"host": opensearch_host, "port": 9200}],
+    "http_compress": True,
+    "http_auth": ("admin", "admin"),
+    "use_ssl": True,
+    "verify_certs": False,
+    "ssl_assert_hostname": False,
+    "ssl_show_warn": False,
+    "timeout": 120,
+}
+
+os_client = OpenSearch(**os_client_args)  # type: ignore
+if os_client.indices.exists(index=INDEX):
+    if args.delete:
+        print(f"Index {INDEX} already exists, deleting as requested")
+        os_client.indices.delete(index=INDEX)
+    else:
+        raise Exception(f"Index {INDEX} already exists. Run with --delete to delete it.")
+
+index_settings = {
+    "body": {
+        "settings": {"index.knn": True, "number_of_shards": 5, "number_of_replicas": 1},
+        "mappings": {
+            "properties": {
+                "embedding": {
+                    "dimension": 384,
+                    "method": {
+                        "engine": "faiss",
+                        "space_type": "l2",
+                        "name": "hnsw",
+                        "parameters": {},
+                    },
+                    "type": "knn_vector",
+                }
+            }
+        },
+    }
+}
+
+
+context = sycamore.init()
+docset = context.read.binary(SOURCE_DATA_PATH, binary_format="pdf")
+tokenizer = HuggingFaceTokenizer("thenlper/gte-small")
+llm = OpenAI(OpenAIModels.GPT_3_5_TURBO.value)
+
+# partitioning docset
+partitioned_docset = (
+    docset.partition(partitioner=UnstructuredPdfPartitioner())
```

Comment:
Can we use ArynPartitioner please

Reply:
Done, but scares me a lot that as soon as I changed the partitioner I got different results.